File: nattybear/level1/정수-내림차순으로-배치하기/solution.js
function solution(n) {
  const xs = digits(n).sort((x, y) => y - x)
  return number(xs)
}

function digits(n) {
  const xs = []
  while (n > 0) {
    xs.push(n % 10)
    n = Math.floor(n / 10)
  }
  return xs
}

function number(xs) {
  const size = xs.length
  let sum = 0
  for (let i = 0; i < size; i++)
    sum += xs[i] * (10 ** (size - i - 1))
  return sum
}

File: nattybear/level1/정수-내림차순으로-배치하기/Solution.java
import java.util.ArrayList;
import java.util.Collections;

class Solution {
  public long solution(long n) {
    ArrayList<Integer> digits = toDigits(n);
    Collections.reverse(digits);
    return toNumber(digits);
  }

  static ArrayList<Integer> toDigits(long n) {
    ArrayList<Integer> digits = new ArrayList<>();

    while (n > 0) {
      int remainder = (int) (n % 10);
      digits.add(remainder);
      n = n / 10;
    }

    return digits;
  }

  static long toNumber(ArrayList<Integer> digits) {
    long sum = 0;

    for (int i = 0; i < digits.size(); i++)
      sum += digits.get(i) * (int) Math.pow(10, digits.size() - i + 1);

    return sum;
  }
}
